sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/LargeBlobTest.groovy

```diff
@@ -28,8 +32,11 @@ class LargeBlobTest extends APISpec {
     BlobClient blobClient
     BlobAsyncClient blobAsyncClient
     String blobName
-    List<Mono<Long>> putBlockPayloadSizes = Collections.synchronizedList(new ArrayList<>())
-    AtomicLong count = new AtomicLong()
+    List<Long> putBlockPayloadSizes = Collections.synchronizedList(new ArrayList<>())
+    AtomicLong blocksCount = new AtomicLong()
+    List<Long> putBlobPayloadSizes = Collections.synchronizedList(new ArrayList<>())
+    AtomicLong singleUploadCount = new AtomicLong()
+    ConcurrentHashMap<String, Boolean> retryTracker = new ConcurrentHashMap<>()
```

Comment:
Should these be test class level properties or be migrated into the pipeline policy being used to test this functionality? I don't know when in the test loop, if ever, these would be reset to the default value again.

Reply:
Every test framework I worked with JUnit, TestNG (Java) or NUnit, XUnit, MSTest (.Net) instantiates new instance of test class for each test case. Therefore these are reset for each test.
I could potentially put them into policy but then I'd need to keep reference to policy in test case.

I'd rather keep them in test class (they represent test case state) rather than in policy (that is used to update test case state).

Reference: https://stackoverflow.com/questions/19381352/does-junit-reinitialize-the-class-with-each-test-method-invocation

Reply:
Good with me, was verifying that no bad state would be reached with these in the class scope.